src/frontend/src/scheduler/plan_fragmenter.rs

```diff
@@ -659,10 +687,12 @@ impl StageGraph {
                 .clone()
                 .complete(self.batch_parallelism)
                 .await?;
-
+            // For file source batch read, all the files  to be read are divide into several parts to prevent the task from taking up too many resources.
+            // todo(wcy-fdu): Currently it will be divided into half of batch_parallelism groups, and this will be changed to configurable later.
             let complete_stage = Arc::new(stage.clone_with_exchange_info_and_complete_source_info(
                 exchange_info,
                 complete_source_info,
+                (self.batch_parallelism / 2) as u32,
```

Comment:
here we fix the overall parallelism to a relatively small one (prev the split number). May result in batch pref regression for kafka batch scan. Leave the review as a reminder

Reply:
I suggest to move this default batch parallelism change from this PR to a separate PR, because it would affect so many current batch performance tests result The performance regression can be identified easier and make it easy to revert if needed.

Reply:
Make sense. Let me open a separate PR and do some perf test.

Reply:
Block by https://github.com/risingwavelabs/risingwave/pull/15416

Reply:
let's distinguish between file source and other source when setting task parallelism.